dask/bytes/core.py

```diff
@@ -34,123 +34,46 @@ def write_block_to_file(data, f, compression, encoding):
         which can be read.
     f : file-like
         backend-dependent file-like object
-    compression : string
-        a key of `compress_files`
-    encoding : string (None)
-        if a string (e.g., 'ascii', 'utf8'), implies text mode, otherwise no
-        encoding and binary mode.
     """
-    original = False
-    f2 = f
-    f = SeekableFile(f)
-    if compression:
-        original = True
-        f = compress_files[compression](f, mode='wb')
+    binary = 'b' in str(getattr(f, 'mode', 'b'))
     try:
         if isinstance(data, (str, bytes)):
-            if encoding:
-                f.write(data.encode(encoding=encoding))
-            else:
-                f.write(data)
+            f.write(data)
         elif isinstance(data, io.IOBase):
             # file-like
             out = '1'
             while out:
                 out = data.read(64 * 2 ** 10)
-                if encoding:
-                    f.write(out.encode(encoding=encoding))
-                else:
-                    f.write(out)
+                f.write(out)
         else:
             # iterable, e.g., bag contents
             start = False
             for d in data:
                 if start:
-                    f.write(b'\n')
+                    if binary:
+                        try:
+                            f.write(b'\n')
+                        except TypeError:
+                            binary = False
+                            f.write('\n')
+                    else:
+                        f.write('\n')
```

Comment:
This is because we might be writing to a text-mode or binary-mode file. I don't know of any other foolproof way to test which the file is.